### PR TITLE
Allow overrides of PACKED_ATTRIBUTE preprocessor macro

### PR DESCRIPTION
**Pull request by rmcdonald**
This prevents libutp code from defining PACKED_ATTRIBUTE if it has already been defined.  This greatly reduces noisy gcc warnings since to override definition - avoids gcc warnings about redefining a preprocessor macro when building client products since this macro is defined elsewhere in client code and has different conditions to its definition than this definition.

@xercesblue @angitbit please review.

### CODE REVIEW

File: utp_types.h [hunk 1]
```diff
@@ -23,6 +23,8 @@
 #ifndef __UTP_TYPES_H__
 #define __UTP_TYPES_H__
 
+// Allow libutp consumers or prerequisites to override PACKED_ATTRIBUTE
+#ifndef PACKED_ATTRIBUTE
 #ifdef __GNUC__
```
Comment (angitbit) [inline utp_types.h]: can you include this too? then lgtm
https://github.com/bt-mobile/libutp/commit/1d10a9c5167782c68214dbcb7bc1c00f8a6622a6

Comment (angitbit) [inline utp_types.h]: that pr was https://github.com/bt-mobile/libutp/pull/1 but never went into main repo

**Comment by rmcdonald** [comment]
Adding @mark-bittorrent @jpcottin @B-C @jeanetteliu as reviewers if interested.

File: utp_types.h (hunk 1 above)
Comment (rmcdonald) [inline utp_types.h]: I thought about adding that, but wasn't sure if anybody else needed it.  OK, I can do this.

Comment (angitbit) [inline utp_types.h]: I don't recall android needing it but you said older gcc does {for other posix's ?)

**Comment by angitbit** [comment]
also can you add this redefine fix? https://github.com/bt-mobile/ut_core/commit/d7b1ad2afb1d50aea4e385822ff120ea01e0cc4f
from https://github.com/bt-mobile/ut_core/pull/10

**Comment by rmcdonald** [comment]
@angitbit this second change is in a different repo.  Maybe that would be OK, too.  My motivation for permitting overrides in this repo was the different conditional applied in ut_core, and since libutp is a public repo, I didn't want to add something that appeared to be used only by BitTorrent developers.  Choosing whether to add an override to this repo, or to use this repo as the sole source of this definition, would be part of this review.  I would rather not have libutp be the sole source of this definition because there may be other products that use something relatively generic like this but that don't include libutp.

**Comment by angitbit** [comment]
you mean you'd rather not cut from StdAfxCore.h? if so shouldn't you add #ifndef PACKED_ATTRIBUTE to it too, to avoid redefine?

**Comment by rmcdonald** [comment]
No, I wanted ut_core's PACKED_ATTRIBUTE to be predominant, to override the one in libutp.  The compiler will gripe if there's already one defined when it hits the one in StdAfxCore.h.

**Comment by angitbit** [comment]
ok, thought you were defining elsewhere

**Comment by jpcottin** [comment]
ok

**Comment by xercesblue** [comment]
lgtm

**Pull request merged**